qpulseaudioengine.cpp

```diff
@@ -312,29 +351,43 @@ void QPulseAudioEngineWorker::sinkInfoCallback(const pa_sink_info *info)
         return;
 
     /* Now to decide which output to be used, depending on the active mode */
+    qDebug("Deciding output...");
     if (m_audiomode & AudioModeEarpiece) {
         preferred = earpiece;
         audiomodetoset = AudioModeEarpiece;
+        qDebug("Prefer AudioModeEarpiece");
     }
     if (m_audiomode & AudioModeSpeaker) {
         preferred = speaker;
         audiomodetoset = AudioModeSpeaker;
+        qDebug("Prefer AudioModeSpeaker");
     }
     if ((m_audiomode & AudioModeWiredHeadset) && (modes.contains(AudioModeWiredHeadset))) {
         preferred = wired_headset ? wired_headset : wired_headphone;
         audiomodetoset = AudioModeWiredHeadset;
+        qDebug("Prefer AudioModeWiredHeadset");
     }
     if (m_callstatus == CallRinging && speaker_and_wired_headphone) {
         preferred = speaker_and_wired_headphone;
+        audiomodetoset = AudioModeWiredOrSpeaker;
+        qDebug("Prefer AudioModeWiredOrSpeaker");
     }
     if ((m_audiomode & AudioModeBluetooth) && (modes.contains(AudioModeBluetooth))) {
         preferred = bluetooth_sco;
         audiomodetoset = AudioModeBluetooth;
+        qDebug("Prefer AudioModeBluetooth");
```

Comment:
I would remove that logging from production code as in general logfiles are just noise on the flash memory and will wear it off ;) - and also log rotation is set so shortly that you probably do not see them after a few days anyways.

Reply:
The service is dbus-activated, not started by upstart, meaning it's not showing up in typical log files. The output only shows when run manually.

Reply:
I must disagree. The wear on the flash is really quite small. For a much worse example of this, do `tail -f /var/log/syslog`, then turn your display on and off.

Reply:
My mistake, dbus logs are stored in `~/.cache/upstart/dbus.log`.